numba/unicode.py

```diff
@@ -745,6 +745,33 @@ def concat_impl(a, b):
         return concat_impl
 
 
+@overload(operator.mul)
+def unicode_repeat(a, b):
+    if isinstance(a, types.UnicodeType) and isinstance(b, types.Integer):
+        def repeat_impl(a, b):
+            new_length = a._length * b
+            new_kind = a._kind
+            result = _empty_string(new_kind, new_length)
+            len_a = len(a)
+            for j in range(b):
+                for i in range(len_a):
+                    _set_code_point(result, len_a*j + i, _get_code_point(a, i))
```

Comment:
```suggestion
                    _set_code_point(result, len_a * j + i, _get_code_point(a, i))
```

Reply:
Also, does the `_strncpy()` function in this module help? It does the same sort of loop but will use `memcpy` if it can. And if you want to get really fancy with performance stuff, I think this can be done in `O(n * log_2 (n))` by doubling the copy amount each time. Like `'x' -> 'xx' -> 'xxxx'` etc.

Reply:
thx, I implemented both using strncpy and the faster copying scheme.